pkgs/applications/video/freetube/default.nix

```diff
@@ -1,51 +1,109 @@
-    name = "${pname}-${version}";
-    inherit src;
+  meta = with lib; {
+    description = "An Open Source YouTube app for privacy";
+    homepage = "https://freetubeapp.io/";
+    license = licenses.agpl3Only;
+    maintainers = with maintainers; [ ryneeverett alyaeanyx ];
+    platforms = [ "x86_64-darwin" "x86_64-linux" "aarch64-darwin" ];
   };
 
-  dontUnpack = true;
-  dontConfigure = true;
-  dontBuild = true;
+  linux = stdenv.mkDerivation rec {
+    inherit pname version src meta;
 
-  nativeBuildInputs = [ makeWrapper ];
+    appimageContents = appimageTools.extractType2 {
+      name = "${pname}-${version}";
+      inherit src;
+    };
 
-  installPhase = ''
-    runHook preInstall
+    dontUnpack = true;
+    dontConfigure = true;
+    dontBuild = true;
 
-    mkdir -p $out/bin $out/share/${pname} $out/share/applications $out/share/icons/hicolor/scalable/apps
+    nativeBuildInputs = [ makeWrapper ];
 
-    cp -a ${appimageContents}/{locales,resources} $out/share/${pname}
-    cp -a ${appimageContents}/freetube.desktop $out/share/applications/${pname}.desktop
-    cp -a ${appimageContents}/usr/share/icons/hicolor/scalable/freetube.svg $out/share/icons/hicolor/scalable/apps
+    installPhase = ''
+      runHook preInstall
 
-    substituteInPlace $out/share/applications/${pname}.desktop \
-      --replace 'Exec=AppRun' 'Exec=${pname}'
+      mkdir -p $out/bin $out/share/${pname} $out/share/applications $out/share/icons/hicolor/scalable/apps
 
-    runHook postInstall
-  '';
+      cp -a ${appimageContents}/{locales,resources} $out/share/${pname}
+      cp -a ${appimageContents}/freetube.desktop $out/share/applications/${pname}.desktop
+      cp -a ${appimageContents}/usr/share/icons/hicolor/scalable/freetube.svg $out/share/icons/hicolor/scalable/apps
 
-  # Electron version is set to 22 in order to match upstream
-  postFixup = ''
-    makeWrapper ${electron_22}/bin/electron $out/bin/${pname} \
-      --add-flags $out/share/${pname}/resources/app.asar
-  '';
+      substituteInPlace $out/share/applications/${pname}.desktop \
+        --replace 'Exec=AppRun' 'Exec=${pname}'
 
-  meta = with lib; {
-    description = "An Open Source YouTube app for privacy";
-    homepage = "https://freetubeapp.io/";
-    license = licenses.agpl3Only;
-    maintainers = with maintainers; [ ryneeverett alyaeanyx ];
-    platforms = [ "x86_64-linux" ];
+      runHook postInstall
+    '';
+
+    # Electron version is set to 22 in order to match upstream
+    postFixup = ''
+      makeWrapper ${electron_22}/bin/electron $out/bin/${pname} \
+        --add-flags $out/share/${pname}/resources/app.asar
+    '';
+
+  };
+  darwin = stdenv.mkDerivation {
+    inherit pname version src meta;
+
+    nativeBuildInputs = [ undmg ];
+
+    sourceRoot = "FreeTube.app";
+
+    installPhase = ''
+      runHook preInstall
+      mkdir -p $out/Applications/FreeTube.app
+      cp -R . $out/Applications/FreeTube.app
+      runHook postInstall
+    '';
```

Comment:
It would be better to use Nixpkgs electron instead of the vendor-provided electron. This way of doing it essentially defeats the purpose of Nix.